[bug]: res_crypto, res_pjsip: compilation fail on 32-bit

### Severity

Blocker

### Versions

20.1.0

### Components/Modules

res_crypto, res_pjsip

### Operating Environment

32-bit

### Frequency of Occurrence

Constant

### Issue Description

Imported from JIRA: https://issues.asterisk.org/jira/browse/ASTERISK-30450

I'm re-reporting this since it was rereported to me yet again. I do not have any proposed fix at this time.

```
  [LD] res_convert.o -> res_convert.so
   [CC] res_crypto.c -> res_crypto.o
res_crypto.c: In function ‘try_load_key.isra’:
res_crypto.c:199:39: warning: ‘%s’ directive output may be truncated writing between 5 and 2147483645 bytes into a region of size 255 [-Wformat-truncation=]
  199 |  snprintf(ffname, sizeof(ffname), "%s/%s", dir, fname);
      |                                       ^~
In file included from /usr/include/stdio.h:867,
                 from /usr/src/asterisk-20.1.0/include/asterisk/compat.h:49,
                 from /usr/src/asterisk-20.1.0/include/asterisk.h:22,
                 from res_crypto.c:34:
/usr/include/i386-linux-gnu/bits/stdio2.h:67:10: note: ‘__builtin___snprintf_chk’ output 7 or more bytes (assuming 2147483647) into a destination of size 256
   67 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   68 |        __bos (__s), __fmt, __va_arg_pack ());
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   [LD] res_crypto.o -> res_crypto.so
   [CC] res_curl.c -> res_curl.o


   [CC] res_pjsip/pjsip_transport_events.c -> res_pjsip/pjsip_transport_events.o
res_pjsip/pjsip_transport_events.c: In function ‘cli_show_monitors’:
res_pjsip/pjsip_transport_events.c:649:18: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 7 has type ‘size_t’ {aka ‘unsigned int’} [-Wformat=]
  649 |   ast_cli(a->fd, " %-46.46s   %-10s   %-9s   %6ld   %8" PRIu64 "   %s\n",
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/asterisk-20.1.0/include/asterisk/compat.h:30,
                 from /usr/src/asterisk-20.1.0/include/asterisk.h:22,
                 from res_pjsip/pjsip_transport_events.c:30:
/usr/include/inttypes.h:105:34: note: format string is defined here
  105 | # define PRIu64  __PRI64_PREFIX "u"
   [CC] res_pjsip/pjsip_transport_management.c -> res_pjsip/pjsip_transport_ma
```

```
res_crypto.c: In function â€˜try_load_key.israâ€™:
res_crypto.c:199:35: error: â€˜%sâ€™ directive output may be truncated writing between 5 and 2147483645 bytes into a region of size 255 [-Werror=format-truncation=]
199 | snprintf(ffname, sizeof(ffname), "%s/%s", dir, fname);
| ^~~~~~~
res_crypto.c:199:2: note: â€˜snprintfâ€™ output 7 or more bytes (assuming 2147483647) into a destination of size 256
199 | snprintf(ffname, sizeof(ffname), "%s/%s", dir, fname);
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Relevant log output

_No response_

### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

--- a/res/res_crypto.c
+++ b/res/res_crypto.c
@@ -196,7 +196,14 @@ static struct ast_key *try_load_key(const char *dir, const char *fname, int ifd,
 	}
 
 	/* Get actual filename */
-	snprintf(ffname, sizeof(ffname), "%s/%s", dir, fname);
+	n = snprintf(ffname, sizeof(ffname), "%s/%s", dir, fname);
+	if (n >= sizeof(ffname)) {
+		ast_log(LOG_WARNING,
+			"Key filenames can be up to %zu bytes long, but the filename for the"
+			" key we are currently trying to load (%s/%s) is %d bytes long.",
+			sizeof(ffname) - 1, dir, fname, n);
+		return NULL;
+	}
 
 	/* Open file */
 	if (!(f = fopen(ffname, "r"))) {

--- a/res/res_pjsip/pjsip_transport_events.c
+++ b/res/res_pjsip/pjsip_transport_events.c
@@ -646,7 +646,7 @@ static char *cli_show_monitors(struct ast_cli_entry *e, int cmd, struct ast_cli_
 			state = "ACTIVE";
 		}
 
-		ast_cli(a->fd, " %-46.46s   %-10s   %-9s   %6ld   %8" PRIu64 "   %s\n",
+		ast_cli(a->fd, " %-46.46s   %-10s   %-9s   %6ld   %8zu   %s\n",
 			monitored->key, state,
 			monitored->transport->dir == PJSIP_TP_DIR_OUTGOING ? "Outgoing" : "Incoming",
 			pj_atomic_get(monitored->transport->ref_cnt),

